### PR TITLE
Feat: Enable removal of 1-epoch datasets

### LINKED ISSUE #1353: Remove temporary FWSS activation-boundary settlement skip after contract upgrade
Curio currently has a temporary workaround in shouldSkipFWSSActivationTerminationBoundary to skip terminated FWSS PDP rails where endEpoch == activationEpoch.
This exists because the current FWSS contract cannot settle that boundary case: validatePayment allows toEpoch == activationEpoch, but the proving-period helper treats that epoch as invalid, leaving Curio with no valid settlement target.
Once the upgraded FWSS contracts are deployed and active, remove this workaround and its tests. Curio should then rely on the contract to handle the activation-boundary case correctly instead of skipping those rails client-side.

This issue should be resolved with a PR once https://github.com/FilOzone/filecoin-services/issues/544 is deployed

### PR DESCRIPTION
**Pull request by snadrus**
Fixes https://github.com/filecoin-project/curio/issues/1353

### CODE REVIEW

**Review by ZenGround0 — APPROVED** [review]

File: pdp/contract/FWSS/settle_resolver_test.go [hunk 1]
```diff
@@ -154,13 +154,14 @@ func TestFWSSPDPSettleTarget(t *testing.T) {
 			wantResolvable: true,
 		},
 		{
-			name:           "terminated rail ending at activation is skipped for FWSS contract bug",
```
Comment (ZenGround0) [inline pdp/contract/FWSS/settle_resolver_test.go:157]: Shouldn't we just be removing this testcase ?

Comment (snadrus) [reply pdp/contract/FWSS/settle_resolver_test.go:157]: It now tests: settledUpTo < endEpoch == activation
